2020/day15/solution.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import  annotations
+
+input = [15,5,1,4,7,0]
+
+def find_nth_number(starting : list[int], nth : int) -> int:
+    last_spoken = { value : [index] for index, value in enumerate(starting)}
+    round = len(starting)
+    last_number = starting[-1]
+    while round < nth:
+        if len(last_spoken[last_number]) == 1:
+            last_number = 0
+        elif len(last_spoken[last_number]) == 2:
+            penultimate, ultimate = last_spoken[last_number]
+            last_number = ultimate - penultimate
+        else:
+            print("OOPS")
+            return -1
```

Comment:
I should probably turn  this into an assert since it means I did something wrong